code/sorting/src/sleep_sort/sleep_sort.c

```diff
@@ -1,34 +1,87 @@
 /* Part of Cosmos by OpenGenus Foundation */
 
-#include <stdio.h>
-#if defined(WIN32) || defined(_WIN32) || defined(__WIN32) && !defined(__CYGWIN__)
-    #include <windows.h>
-    #include <process.h>
-#endif
+/* sleep sort works both in windows and linux*/
 
+/* WINDOWS SPECIFIC CODE*/
+#if defined(WIN32) || defined(_WIN32) || \
+    defined(__WIN32) && !defined(__CYGWIN__)
+#include <process.h>
+#include <windows.h>
 void routine(void *a)
 {
-    int n = *(int *) a;
-    Sleep(n);
-    printf("%d ", n);
+  int n = *(int *)a;
+  Sleep(n);
+  printf("%d ", n);
 }
 
 void sleepSort(int arr[], int n)
 {
-    int i;
-    HANDLE threads[n];
-    for (i = 0; i < n; i++)
-        threads[i] = (HANDLE)_beginthread(&routine, 0, &arr[i]);
-    WaitForMultipleObjects(n, threads, TRUE, INFINITE);
-    return;
+  int i;
+  HANDLE threads[n];
+  for (i = 0; i < n; i++)
+    threads[i] = (HANDLE)_beginthread(&routine, 0, &arr[i]);
+  WaitForMultipleObjects(n, threads, TRUE, INFINITE);
+  return;
+}
+
+/* LINUX SPECIFIC CODE*/
+#elif defined(__linux__)
+#include <assert.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+
+static void fdpipe(FILE** readFp, FILE** writeFp)
+{
+  int fds[2], err;
```

Comment:
Why is `err` declared here? We use modern C, so just declare it in the middle of the function (i.e `int err = pipe(fds)`).